lib/ts/recipe/passwordless/components/features/signInAndUp/index.tsx

```
@@ -264,81 +267,112 @@ export const SignInUpFeature: React.FC<
+                    const loginAttemptInfo = await getLoginAttemptInfo({
+                        recipeImplementation: recipe.recipeImpl,
+                        userContext: input.userContext,
+                    });
+                    // If it was cleared or overwritten we don't want to save this.
+                    // TODO: extend session checker to check for this case as well
+                    if (loginAttemptInfo !== undefined && loginAttemptInfo.deviceId === input.deviceId) {
+                        const timestamp = Date.now();
+
+                        await setLoginAttemptInfo({
+                            recipeImplementation: recipe.recipeImpl,
+                            userContext: input.userContext,
+                            attemptInfo: {
+                                ...loginAttemptInfo,
+                                lastResend: timestamp,
+                            },
+                        });
+                        dispatch({ type: "resendCode", timestamp });
+                    }
+                } else if (res.status === "RESTART_FLOW_ERROR") {
+                    await recipe.recipeImpl.clearLoginAttemptInfo({
+                        userContext: input.userContext,
+                    });
+
+                    dispatch({ type: "restartFlow", error: "ERROR_SIGN_IN_UP_RESEND_RESTART_FLOW" });
+                }
+                return res;
+            } catch (e) {
+                if (STGeneralError.isThisError(e)) {
+                    dispatch({ type: "setError", error: e.message });
                 }
-            } else if (res.status === "RESTART_FLOW_ERROR") {
-                await originalImpl.clearLoginAttemptInfo();
 
-                dispatch({ type: "restartFlow", error: "ERROR_SIGN_IN_UP_RESEND_RESTART_FLOW" });
-            } else if (res.status === "GENERAL_ERROR") {
-                dispatch({ type: "setError", error: res.message });
+                throw e;
```

Do you want to throw an error even if `STGeneralError.isThisError(e)` is `true`?

---

Yes, this is the modified recipe implementation. When this function is called the errors get handled there.